src/impala/sema/infersema.cpp

```diff
@@ -757,7 +757,9 @@ const Type* FieldExpr::check(InferSema& sema) const {
             return struct_type->op(field_decl->index());
     }
 
-    return sema.type_error();
+    return ltype->is_known()
+        ? static_cast<const Type*>(sema.type_error())
+        : static_cast<const Type*>(sema.unknown_type());
```

Comment:
It does not: `ltype` is the *structure type* in the expression `struct.member`. Hence you do not want to assign the same type to the result of the member access. Maybe this would do what you want:
```cpp
return ltype->is_known()
    ? sema.type_error()
    : (!type_ || !type_->is_known() ? sema.unknown_type() : type_);
```